STYLEGUIDE.md

```diff
@@ -0,0 +1,367 @@
+ */
+export const getNextUpgradeInstance = (
+  clusterDescription: ClusterDescription,
+  providerOptionsModel: ProviderOptions,
+) => {
+  // ...
+};
+```
+
+#### Avoid
+
+```typescript
+export const getNextUpgradeInstance = (
+  clusterDescription: ClusterDescription,
+  providerOptionsModel: ProviderOptions,
+) => {
+  // ...
+};
+```
+
+---
+
+### Prefer creating the function in the highest scope, and passing in all required data.
+
+#### Why
+
+Keeps functions pure.
+
+---
+
+## <a id="Variable-naming">Variable Naming</a>
+
+### Acronyms are considered a single entity in camel-casing
+
+#### Why
+
+Some treat each letter in an acronym as its own entity when an acronyme is included in a var name. We'll instead consider the acronym as a single entity, and only capitalize the first letter of it (if that).
+
+#### Prefer
+
+```typescript
+let myWipValue = 'pending';
+```
+
+#### Avoid
+
+```typescript
+let myWIPValue = 'pending';
+```
+
+---
+
+### Use UPPERCASE for static constants, and refactor to a shared file if needed.
+
+---
+
+### Use camelCase for most javascript functions and variables
+
+---
+
+### Use “be” verbs for boolean types. (e.g. should`/`can`/`does`/`has`)
+
+---
+
+## <a id="Variable-placement">Variable placement</a>
+
+### Avoid inline declaration of static variables when they can be moved to top of file or refactored into new file
+
+---
+
+### Prefer placing let/const variables at the beginning of their scope.
+
+---
+
+### Prefer const to let whenever possible.
+
+---
+
+## <a id="Whitespace-and-line-breaks">Whitespace and line breaks</a>
+
+### Use a space before every prop defined in an interface
+
+---
+
+## <a id="JSX-React">JSX/React</a>
+
+### Prefer factoring out any more-than-1-line event handlers into their own functions
+
+#### Why
+
+Minimizes inline javascript logic within
+
+#### Prefer
+
+```typescript
+const handleClick = (e) => {
+    e.preventDefault()
+    setState(curr => !curr)
+}
+
+<button onClick={handleClick}>
+```
+
+#### Avoid
+
+```typescript
+<button onClick={(e) => {
+    e.preventDefault()
+    setState(curr => !curr)
+}}>
+```
+
+---
+
+### Prefer prefixing event handlers with `handle`
+
+#### Why
+
+Standardizes how we name and search for functions that handle events
+
+#### Prefer
+
+```typescript
+const handleClick = () => {};
+return <button onClick={handleClick} />;
+```
+
+#### Avoid
+
+```typescript
+const onClick = () => {};
+return <button onClick={onClick} />;
+```
+
+---
+
+### Prefer using fragments over divs whenever possible
+
+#### Why
+
+Consolidates the number of levels of DOM
+
+#### Prefer
+
+```typescript
+return <></>;
+```
+
+#### Avoid
+
+```typescript
+return <div></div>;
+```
+
+---
+
+### Prefer using spread operator to avoid mutation of arrays and objects
+
+#### Why
+
+Immutability
+
+---
+
+### Avoid nesting ternary operators
+
+---
+
+### Factor out of functional components any methods that do NOT require state
```

Comment:
Feels hard to parse. Would change to `"Factor out any methods that do NOT require state from functional components"`,  or even `"Prefer utility functions when full component state is not required"`